pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -185,6 +185,41 @@ func makeMounts(cid string, opts *kubecontainer.RunContainerOptions, container *
 	return volumeMounts
 }
 
+// getKubeletContainers lists all (or just the running) containers managed by kubelet.
+func (m *kubeGenericRuntimeManager) getKubeletContainers(allContainers bool) ([]*runtimeApi.Container, error) {
+	var resp []*runtimeApi.Container
+	var err error
+
+	if allContainers {
+		resp, err = m.runtimeService.ListContainers(nil)
+	} else {
+		runningState := runtimeApi.ContainerState_RUNNING
+		resp, err = m.runtimeService.ListContainers(&runtimeApi.ContainerFilter{
+			State: &runningState,
+		})
+	}
+	if err != nil {
+		glog.Errorf("ListContainers failed: %v", err)
+		return nil, err
+	}
+
+	result := []*runtimeApi.Container{}
+	for _, c := range resp {
+		if len(c.GetName()) == 0 {
```

Comment:
hm...why would we have a container with no name, and shouldn't this be handled by the `if` below anyway?